### PR TITLE
Contacts export for exchange

### PR DESCRIPTION
**Pull request by meain**
---

#### Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [ ] :no_entry: No

#### Type of change

- [x] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)

* closes https://github.com/alcionai/corso/issues/3891

#### Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/internal/converters/vcf/vcf.go
Comment (ashmrtn) [inline src/internal/converters/vcf/vcf.go:1]: do we have an idea of what the range of valid inputs are? For example, is it possible that we'll get malformed emails, phone numbers, or addresses from the source data that could cause problems for the vcard code?

**Review by ashmrtn — COMMENTED** [review]
generally looks good. Would like to see some info discussion around possible invalid inputs (if any) and some clarity/testing around the organization field

File: src/internal/converters/vcf/vcf.go [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+		if len(orgFull) > 0 {
+			orgFull += ";"
+		}
+
+		orgFull += *profession
+	}
+
+	if len(orgFull) > 0 {
+		vc.Set(vcard.FieldOrganization, &vcard.Field{Value: orgFull})
+	}
```
[the remark this answers is not in the document]
Comment (meain) [reply src/internal/converters/vcf/vcf.go:157]: We will set the other fields even if the org name is not available. The `if` condition pretty much decides if we prefix a `;`. That said, now that I think about I think we should have an empty column to symbolize that the org name is empty.

The [spec](https://datatracker.ietf.org/doc/html/rfc6350) does not go into any detail around what we should do if the name is not available.

Also, let me add the necessary tests.

File: src/internal/converters/vcf/vcf.go
Comment (meain) [reply src/internal/converters/vcf/vcf.go:1]: Unlike in the case of emails, the vcard lib that we currently does not use any validation on the strings that pass to it and so whatever is the data that we get from graph goes into the output. Most values are free-form text. `TEL` is "free-form text value" in spec and `EMAIL` is "free-form UTF-8 text" and so I don't think it matter as far as spec is concerned.

**Review by ashmrtn — APPROVED** [review]

**Pull request merged**